src/DynamoSandbox/DynamoCoreSetup.cs

```diff
@@ -145,6 +136,89 @@ public void RunApplication(Application app)
             }
         }
 
+        private async void ImportSettings(string fileContent)
+        {
+            var fileName = string.Empty;
+
+            using (FileStream fs = File.Create("preferenceSettings.xml"))
+            {
+                Byte[] title = new UTF8Encoding(true).GetBytes(fileContent);
+                fs.Write(title, 0, title.Length);
+                fileName = fs.Name;
+            }
+
+            var importResult = viewModel.PreferencesViewModel.importSettings(fileName);
+            if (importResult)
+            {
+                migrationWindow.SetImportStatus(ImportStatus.success, Resources.SplashScreenSettingsImported, string.Empty);
+            }
+            else
+            {
+                migrationWindow.SetImportStatus(ImportStatus.error, Resources.SplashScreenFailedImportSettings, Resources.SplashScreenImportSettingsFailDescription);
+            }
+        }
+
+        private void DynamoModel_RequestUpdateLoadBarStatus(SplashScreenEventArgs args)
+        {
+            if(migrationWindow != null)
+            {
+                migrationWindow.SetBarProperties(Dynamo.Utilities.AssemblyHelper.GetDynamoVersion().ToString(),
+                    args.LoadDescription, args.BarSize);
+            }
+        }
+
+        private void LoadDynamoView()
```

Comment:
This is great but I anticipate us moving this code to DynamoCoreWpf so that this can be shared between sandbox and integrations.. I would like to discuss with you tomorrow about it